src/mbgl/layout/symbol_layout.cpp

```diff
@@ -572,23 +572,47 @@ void SymbolLayout::addFeature(const std::size_t layoutFeatureIndex,
         }
     }
 
-    auto addSymbolInstance = [&] (Anchor& anchor, std::shared_ptr<SymbolInstanceSharedData> sharedData) {
+    auto addSymbolInstance = [&](Anchor& anchor, std::shared_ptr<SymbolInstanceSharedData> sharedData) {
         assert(sharedData);
-        const bool anchorInsideTile = anchor.point.x >= 0 && anchor.point.x < util::EXTENT && anchor.point.y >= 0 && anchor.point.y < util::EXTENT;
+        const bool anchorInsideTile = anchor.point.x >= 0 && anchor.point.x < util::EXTENT && anchor.point.y >= 0 &&
+                                      anchor.point.y < util::EXTENT;
 
         if (mode == MapMode::Tile || anchorInsideTile) {
+            if (sortFeaturesByKey) {
+                if (sortKeyRanges.size() && sortKeyRanges.back().sortKey == feature.sortKey) {
+                    sortKeyRanges.back().symbolInstanceEnd = symbolInstances.size() + 1;
```

Comment:
`sortKeyRanges.back().symbolInstanceEnd = symbolInstances.size();` shall be enough, no? `begin()` + size = `end()` :thinking:

Reply:
It looks like I'm setting it before adding the new symbol instance. I guess moving it after and removing the `+ 1` would be cleaner